torch/csrc/jit/script/concrete_module_type.cpp

```diff
@@ -27,6 +70,17 @@ ClassTypePtr ConcreteModuleTypeBuilder::createTypeFromThis() const {
     cls->addAttribute(name, type, isParameter);
   }
 
+  for (const auto& pr : constants_) {
+    const auto& name = pr.first;
+    const auto& val = pr.second.v_;
+    auto match = tryToInferType(val);
+    if (!match.success()) {
+      TORCH_CHECK(false, "Can't infer type of ", py::str(val), "\n:", match.reason());
```

Comment:
This is not a great error message. It doesn't describe why we need to infer the type of this value.